src/query/functions/src/scalars/geometry.rs

```diff
@@ -1034,7 +1036,103 @@ pub fn register(registry: &mut FunctionRegistry) {
                     }
                 };
 
-                match st_xmax(&geo) {
+                match st_extreme(&geo, Axis::X, Extremum::Max) {
+                    None => builder.push_null(),
+                    Some(x_max) => builder.push(F64::from(AsPrimitive::<f64>::as_(x_max))),
+                };
+            },
+        ),
+    );
+
+    registry.register_combine_nullable_1_arg::<GeometryType, NumberType<F64>, _, _>(
+        "st_xmin",
+        |_, _| FunctionDomain::MayThrow,
+        vectorize_with_builder_1_arg::<GeometryType, NullableType<NumberType<F64>>>(
+            |geometry, builder, ctx| {
+                if let Some(validity) = &ctx.validity {
+                    if !validity.get_bit(builder.len()) {
+                        builder.push_null();
+                        return;
+                    }
+                }
+
+                let geo: geo_types::Geometry = match Ewkb(geometry).to_geo() {
```

Comment:
We have those code in every function that converts binary geometry to `Geometry` struct, and since all `Geometry` data is checked when created or inserted to tables, can we assume that this conversion is always successful? We can remove the error handling codes to make the function more clear, and the domain returned by the function could be changed from `FunctionDomain::MayThrow` to `FunctionDomain::Full` or a specific value range. Domain can helps us to filter the range when querying.